sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/RegexpExpressionsSuite.scala

```diff
@@ -391,12 +391,12 @@ class RegexpExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     val row11 = create_row("100-200", "(\\d+)-(\\d+)", -1)
     val row12 = create_row("100-200", "\\d+", -1)
 
-    checkExceptionInExpression[IllegalArgumentException](
-      expr, row8, "Regex group count is 2, but the specified group index is 3")
-    checkExceptionInExpression[IllegalArgumentException](
-      expr, row9, "Regex group count is 1, but the specified group index is 2")
-    checkExceptionInExpression[IllegalArgumentException](
-      expr, row10, "Regex group count is 0, but the specified group index is 1")
+    checkExceptionInExpression[SparkRuntimeException](
```

Comment:
Could you use `checkErrorInExpression()` instead of `checkExceptionInExpression()` to don't depend on the error message anymore, please.